packages/order-experience-api/src/api/createRefund.ts

```diff
@@ -187,7 +186,7 @@ export class CreateRefundController extends AbstractController<
               items: refundItems,
             },
             payment,
-            gateway: RefundGateway.PAYTRAIL.toString(),
+            gateway: payment.paymentGateway,
```

Comment:
Default to RefundGateway.PAYTRAIL.toString()

Use like `payment.paymentGateway || RefundGateway.PAYTRAIL.toString()`